tests/test_nystrom_attention.py

```diff
@@ -85,8 +85,10 @@ def test_att_mask_ignored():
         assert torch.allclose(r_nystrom, r_sdp, rtol=0.005, atol=1e-2)
 
     def test_masking():
-        nystrom_config["causal"] = True
-        sdp_config["causal"] = True
+        # FIXME: Masking seems fine as such, but whole lines are randomly masked when causal+random mask
```

Comment:
Is this what you meant by key padding + causal being broken?

Reply:
yes, although it feels like it could be only an issue with the unit test, but scaled_dot_product and nystrom are not behaving in the same way (although there are plenty of NaNs around in both cases). I think that it could be that Nystrom breaks earlier because of the pooling for anchors, if one anchor is full of inf (easy with causal) it cannot compute the attention

Reply:
Fixed now with a random masking a little less aggressive (not masking half of the words :)), I think that it's still a good test

Reply:
Nice :) So can causal be used for the test now or that's still an issue?

Reply:
not an issue anymore, I messed up the push when rebasing but it works fine locally, let me update..